app/src/main/java/com/example/tasky/feature_authentication/presentation/login_screen/LoginScreen.kt

```diff
@@ -0,0 +1,182 @@
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavController
+import com.example.tasky.R
+import com.example.tasky.feature_authentication.domain.util.AuthUseCaseResult
+import com.example.tasky.feature_authentication.domain.validation.EmailError
+import com.example.tasky.feature_authentication.domain.validation.PasswordError
+import com.example.tasky.feature_authentication.presentation.components.TextFieldComponent
+import com.example.tasky.presentation.theme.BackgroundBlack
+import com.example.tasky.presentation.theme.BackgroundWhite
+import com.example.tasky.presentation.theme.BtnNavRegScreen
+import com.example.tasky.presentation.theme.LoginBtnTextColor
+import com.example.tasky.presentation.theme.HintColor
+import com.example.tasky.util.Screen
+
+@Composable
+fun LoginScreen(
+    navController: NavController,
+    viewModel: LoginViewModel = hiltViewModel()
+) {
+    val state by viewModel.state.collectAsState()
+    val context = LocalContext.current
+
+    LaunchedEffect(key1 = viewModel) {
+        viewModel.authResult.collect { result ->
+            when(result) {
+                is AuthUseCaseResult.ErrorEmail -> {
+                    when(result.emailError) {
+                        EmailError.EMAIL_EMPTY -> Toast.makeText(context, R.string.EMAIL_EMPTY, Toast.LENGTH_LONG).show()
+                        EmailError.EMAIL_INVALID -> Toast.makeText(context, R.string.EMAIL_INVALID, Toast.LENGTH_LONG).show()
+                    }
+                }
+                is AuthUseCaseResult.ErrorFullName -> Unit
+                is AuthUseCaseResult.ErrorPassword -> {
+                    when(result.passwordError) {
+                        PasswordError.PASSWORD_EMPTY -> Toast.makeText(context, R.string.PASSWORD_EMPTY, Toast.LENGTH_LONG).show()
+                        PasswordError.PASSWORD_INVALID -> Toast.makeText(context, R.string.PASSWORD_INVALID, Toast.LENGTH_LONG).show()
+                        PasswordError.PASSWORD_TOO_SHORT -> Toast.makeText(context, R.string.PASSWORD_TOO_SHORT, Toast.LENGTH_LONG).show()
+                    }
+                }
+                is AuthUseCaseResult.GenericError -> {
+                    Toast.makeText(context, result.message, Toast.LENGTH_LONG).show()
+                }
+                is AuthUseCaseResult.Success -> {
+                    TODO("Navigate to Agenda Screen")
+                }
+            }
+        }
```

Comment:
From a usability perspective, it would make more sense to have persistent error messages for validated text fields. If the text field's border and text turns red and the error text appears below the text field, it's much clearer for the user what's wrong. 
In your case, only one errors gets shown at a time